activerecord/test/cases/adapters/postgresql/network_test.rb

```diff
@@ -93,4 +93,18 @@ def test_schema_dump_with_shorthand
     assert_match %r{t\.cidr\s+"cidr_address",\s+default: "192\.168\.1\.0/24"}, output
     assert_match %r{t\.macaddr\s+"mac_address",\s+default: "ff:ff:ff:ff:ff:ff"}, output
   end
+
+  def test_upcase_not_dirty
+    mac_addr = "ab:cd:ef:01:02:03"
+    model = PostgresqlNetworkAddress.create(mac_address: mac_addr)
+    model.mac_address = mac_addr.swapcase
+    assert_not model.changed?
+  end
+
+  def test_downcase_not_dirty
+    mac_addr = "AB:CD:EF:01:02:03"
+    model = PostgresqlNetworkAddress.create(mac_address: mac_addr)
+    model.mac_address = mac_addr.swapcase
+    assert_not model.changed?
+  end
```

Comment:
This is a nitpick, and someone else may have a different opinion, but it might be overkill to have two separate tests for this.  If you use a mixed-case `mac_addr` (e.g. `"Ab:cD:Ef:01:02:03"`), you can combine these two tests into `test_case_insensitive` (or similarly named).  That's why I think `swapcase` is so handy. :)

Reply:
Test cases combined into a single swapcase.